### PR TITLE
unittest for Text

### PR DESCRIPTION
**Pull request by hokimcuc0202**

### CODE REVIEW

File: src/react/Text/__tests__/disable.test.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    console.error.mockRestore();
+  });
+
+  test('should be invisible', ()=>{
+    const onChange = (value: string) => {};
+    const onClick = (e: any) => {};
+
+    const {container} = render(<Text value="error" isDisabled isVisible={false} onChange={onChange} onClick={onClick} />);
+    expect(container.firstElementChild).toBeNull();
+  });
```
Comment (phongnhat19) [inline src/react/Text/__tests__/disable.test.tsx]: Should test if the Text component is disabled or not.

File: src/react/Text/__tests__/onEvent.test.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    jest.spyOn(console, 'error');
+    // @ts-ignore
+    console.error.mockImplementation(() => { });
+  });
+  afterEach(() => {
+    // @ts-ignore
+    console.error.mockRestore();
+  });
+
+  test('should be fire onChange event', ()=>{
```
Comment (phongnhat19) [inline src/react/Text/__tests__/onEvent.test.tsx]: Change description to `onChange event handler should fire successfully`

File: src/react/Text/__tests__/onEvent.test.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    const onChange = (value: string) => {
+      expect(value).toBe('hello');
+    };
+    const {container} = render(<Text value="error" onChange={onChange} />);
+    const node = container.getElementsByClassName('kuc-input-text')[0];
+    fireEvent.change(node, {target: {value: 'hello'}});
+  });
+
+  test('should be fire onClick event', ()=>{
```
Comment (phongnhat19) [inline src/react/Text/__tests__/onEvent.test.tsx]: Change description to `onClick event handler should fire successfully`

File: src/react/Text/__tests__/onEvent.test.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    const onChange = (value: string) => {
+      expect(value).toBe('hello');
+    };
```
Comment (phongnhat19) [inline src/react/Text/__tests__/onEvent.test.tsx]: `onChange` event handler is not needed in `onClick` test.

File: src/react/Text/__tests__/onEvent.test.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    const onClick = (e: React.SyntheticEvent<EventTarget, Event>) => {
+      expect((e.target as HTMLInputElement).value).toBe('hello');
+    };
+    const {container} = render(<Text value="error" onChange={onChange} onClick={onClick} />);
+    const node = container.getElementsByClassName('kuc-input-text')[0];
+    fireEvent.change(node, {target: {value: 'hello'}});
```
Comment (phongnhat19) [inline src/react/Text/__tests__/onEvent.test.tsx]: `onChange` event handler is not needed in `onClick` test.

File: src/js/Text/__tests__/rerender.test.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    // @ts-ignore
+    console.error.mockRestore();
+  });
+
+  test('should setValue() null for rerender successfully', ()=>{
+    const text = new Text({value: 'kintone'});
+    // @ts-ignore
+    text.setValue(null);
+    text.rerender();
+    expect(true).toBeTruthy();
```
[the remark this answers is not in the document]
Comment (minhhy2801) [reply src/js/Text/__tests__/rerender.test.ts]: Screen Shot 2020-02-03 at 10 07 37

File: src/react/Text/__tests__/disable.test.tsx [hunk 7]
```diff
@@ -14,12 +14,21 @@ describe('<Text/>', () => {
     console.error.mockRestore();
   });
 
-  test('should be invisible', ()=>{
+  test('should be disabled', ()=>{
```
Comment (phongnhat19) [inline src/react/Text/__tests__/disable.test.tsx]: Update test description

File: src/react/Text/__tests__/disable.test.tsx [hunk 8]
```diff
@@ -14,12 +14,21 @@ describe('<Text/>', () => {
...
     const onChange = (value: string) => {};
     const onClick = (e: any) => {};
```
Comment (phongnhat19) [inline src/react/Text/__tests__/disable.test.tsx]: Remove unnecessary event handler

File: src/react/Text/__tests__/onEvent.test.tsx [hunk 9]
```diff
@@ -14,22 +14,16 @@ describe('<Text/>', () => {
     console.error.mockRestore();
   });
 
-  test('should be fire onChange event', ()=>{
-    const onChange = (value: string) => {
-      expect(value).toBe('hello');
-    };
+  test('onChange event handler should fire successfully', ()=>{
+    const onChange = (value: string) => {};
```
Comment (phongnhat19) [inline src/react/Text/__tests__/onEvent.test.tsx]: Should expect `value` to be correct.

File: src/react/Text/__tests__/onEvent.test.tsx [hunk 10]
```diff
@@ -14,22 +14,16 @@ describe('<Text/>', () => {
...
-    };
-    const onClick = (e: React.SyntheticEvent<EventTarget, Event>) => {
-      expect((e.target as HTMLInputElement).value).toBe('hello');
-    };
+  test('onClick event handler should fire successfully', ()=>{
+    const onChange = (value: string) => {};
+    const onClick = (e: any) => {};
     const {container} = render(<Text value="error" onChange={onChange} onClick={onClick} />);
     const node = container.getElementsByClassName('kuc-input-text')[0];
     fireEvent.change(node, {target: {value: 'hello'}});
```
Comment (phongnhat19) [inline src/react/Text/__tests__/onEvent.test.tsx]: Unnecessary event `onChange` check.

File: src/react/Text/__tests__/onEvent.test.tsx [hunk 11]
```diff
@@ -14,22 +14,16 @@ describe('<Text/>', () => {
...
 
-  test('should be fire onClick event', ()=>{
-    const onChange = (value: string) => {
-      expect(value).toBe('hello');
-    };
-    const onClick = (e: React.SyntheticEvent<EventTarget, Event>) => {
-      expect((e.target as HTMLInputElement).value).toBe('hello');
-    };
+  test('onClick event handler should fire successfully', ()=>{
+    const onChange = (value: string) => {};
```
Comment (phongnhat19) [inline src/react/Text/__tests__/onEvent.test.tsx]: Unnecessary event handler.

**Pull request merged**